tracing-subscriber/src/fmt/time.rs

```diff
@@ -96,6 +96,61 @@ impl FormatTime for Uptime {
     }
 }
 
+/// Print the current UTC time using the [RFC 3339](https://tools.ietf.org/html/rfc3339)
+/// format (for example, 2019-10-16T03:16:56.992060752+00:00).
+///
+/// Note: RFC 3339 is a subset of [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)
+#[cfg(feature = "chrono")]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Default)]
+pub struct Rfc3339;
```

Comment:
Should this default to UTC or local time?

Reply:
Probably local time? Though I could be convinced otherwise.

Reply:
Arguably UTC is better for server logs anyway so you don't have to compare timezones. If someone wants RFC 3339-style timestamps with local time, they can also use `ChronoLocal`.

I can also split this into `Rfc3339Local` and `Rfc3339Utc` but it seems a tad excessive (and the names look super ugly with the long number + title case :wink:)

Reply:
I tried for a while to implement it like you said but ran into annoying type issues. I ended up making it 2 structs, each with a method to use a custom format string instead of the RFC 3339 format.

For the record, the issues I ran into were:
- `chrono::Utc` and `chrono::Local` both have the `now` function, but it's not part of a trait so you need to implement another trait to encapsulate the two
- Both functions return a `chrono::DateTime<chrono::TimeZone>`. However, in order to use the `to_rfc339` and `format` methods, the object needs to be `chrono::DateTime<chrono::Offset>`
- `chrono::offset::Utc` does implement `chrono::Offset` but `chrono::offset::Local` annoyingly does not. Instead, the `Offset` type for `chrono::Local::now()` is `chrono::offset::FixedOffset`